### PR TITLE
Issue-8347 - added new product attribute documentation

### PR DESCRIPTION
**Pull request by dineshvb**
## Purpose of this pull request

This pull request (PR) covers #8347 

## Affected DevDocs pages

-  /videos/fundamentals/add-new-product-attribute.html

whatsnew
Added the 'Product Attribute Option Creation' section to the [How to Add a New Product Attribute](https://devdocs.magento.com/videos/fundamentals/add-new-product-attribute/)  topic.

### CODE REVIEW

File: src/_videos/fundamentals/add-new-product-attribute.md [hunk 1]
```diff
@@ -337,3 +337,44 @@ backend model has executed successfully, so now we’ll set it to Wool and save
 
 Having saved the product, we’ll now move to the frontend.
 It should be visible and in bold text.
+
+## Product Attribute Option Creation
+
+A product attribute of type multiselect or select will present selectable options to the user. These options may be added manually through the admin panel, or by upgrade script. The script process is slightly different depending on whether the options are being added at the moment of attribute creation or whether the options are being added at a later time to an existing attribute.
```
Comment (BarnyShergold) [inline src/_videos/fundamentals/add-new-product-attribute.md:343]:
```suggestion
A product attribute of type multiselect or select will present selectable options to the user. These options may be added manually through the admin panel or by upgrade script. The script process is slightly different depending on whether the options are being added at the moment of attribute creation or whether the options are being added at a later time to an existing attribute.
```

File: src/_videos/fundamentals/add-new-product-attribute.md [hunk 2]
```diff
@@ -337,3 +337,44 @@ backend model has executed successfully, so now we’ll set it to Wool and save
...
+
+### Add options to a new prouduct attribute {#AddOptionsAlongNewProductAttribute}
+
+Basic instructions for creating a product attribute by setup or upgrade script can be found [above](#CreateProductAttributeByUpgradeScript). Before scripting the attribute creation, pick one of these two use cases for your options:
+
+*  You want a set of options which cannot be modified by a user through the admin panel, and which can only be changed through a future code push.
+*  You want a set of options which can be modified, added, or deleted through the admin panel.
```
Comment (BarnyShergold) [inline src/_videos/fundamentals/add-new-product-attribute.md:350]:
```suggestion
1.  You want a set of options which can be modified, added or deleted through the admin panel.
```

File: src/_videos/fundamentals/add-new-product-attribute.md [hunk 3]
```diff
@@ -337,3 +337,44 @@ backend model has executed successfully, so now we’ll set it to Wool and save
...
+
+## Product Attribute Option Creation
+
+A product attribute of type multiselect or select will present selectable options to the user. These options may be added manually through the admin panel, or by upgrade script. The script process is slightly different depending on whether the options are being added at the moment of attribute creation or whether the options are being added at a later time to an existing attribute.
+
+### Add options to a new prouduct attribute {#AddOptionsAlongNewProductAttribute}
+
+Basic instructions for creating a product attribute by setup or upgrade script can be found [above](#CreateProductAttributeByUpgradeScript). Before scripting the attribute creation, pick one of these two use cases for your options:
+
+*  You want a set of options which cannot be modified by a user through the admin panel, and which can only be changed through a future code push.
```
Comment (BarnyShergold) [inline src/_videos/fundamentals/add-new-product-attribute.md:349]: Change to numbered as each are referred to this way further down
```suggestion
1.  You want a set of options which cannot be modified by a user through the admin panel and which can only be changed through a future code push.
```

File: src/_videos/fundamentals/add-new-product-attribute.md [hunk 4]
```diff
@@ -337,3 +337,44 @@ backend model has executed successfully, so now we’ll set it to Wool and save
...
+*  You want a set of options which can be modified, added, or deleted through the admin panel.
+
+In the case of use case 1 (an 'immutable' set of options), follow the above instructions ["Add a source model"](#AddSourceModel). You will create a model that contains and dynamically returns the attribute's selectable options to the client.
```
Comment (BarnyShergold) [inline src/_videos/fundamentals/add-new-product-attribute.md:352]:
```suggestion
For use case `1` (an 'immutable' set of options), follow the above instructions ["Add a source model"](#AddSourceModel). You will create a model that contains and dynamically returns the attribute's selectable options to the client.
```

File: src/_videos/fundamentals/add-new-product-attribute.md [hunk 5]
```diff
@@ -337,3 +337,44 @@ backend model has executed successfully, so now we’ll set it to Wool and save
...
+
+In use case 2 (a 'mutable' set of options), see ["EAV and extension attributes"]({{ site.baseurl }}/guides/v2.4/extension-dev-guide/attributes.html). Make sure to declare 'Magento\Eav\Model\Entity\Attribute\Source\Table' as the value for the 'source' attribute option. This ensures that Magento will store options in the appropriate database table.
```
Comment (BarnyShergold) [inline src/_videos/fundamentals/add-new-product-attribute.md:354]:
```suggestion
For use case `2` (a 'mutable' set of options), see ["EAV and extension attributes"]({{ site.baseurl }}/guides/v2.4/extension-dev-guide/attributes.html). Make sure to declare 'Magento\Eav\Model\Entity\Attribute\Source\Table' as the value for the 'source' attribute option. This ensures that Magento will store options in the appropriate database table.
```

File: src/_videos/fundamentals/add-new-product-attribute.md [hunk 6]
```diff
@@ -337,3 +337,44 @@ backend model has executed successfully, so now we’ll set it to Wool and save
...
+
+With `\Magento\Eav\Setup\EavSetup.php::addAttribute()` and `\Magento\Eav\Setup\EavSetup.php::addAttributeOptions()` you can add a series of options with the following array:
+
```
Comment (BarnyShergold) [inline src/_videos/fundamentals/add-new-product-attribute.md:357]: Would suggest enclosing the two array segments in ```php for formatting clarity

File: src/_videos/fundamentals/add-new-product-attribute.md [hunk 7]
```diff
@@ -337,3 +337,44 @@ backend model has executed successfully, so now we’ll set it to Wool and save
...
+### Add options to an existing product attribute
+
+*  To add options to an 'immutable' set of options, modify the custom source model with the additional options you wish to provide.
+
+*  Adding options to a 'mutable' set of options leverages the same `EavSetup` object as you use when creating an attribute with options, but requires an additional step because `EavSetup` needs to know to which attribute you want to assign new options.
+
+1. Assign an array of new options to a variable:
+
+   `$options = ['attribute_id' => null, 'values' => 'Option 1', 'Option 2', etc]];`
```
Comment (BarnyShergold) [inline src/_videos/fundamentals/add-new-product-attribute.md:372]: Would suggest enclosing the three code segments in ```php for formatting clarity

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Great addition! Just a few formatting & punctuation suggestions

**Comment by BarnyShergold** [comment]
@magento run all tests

File: src/_videos/fundamentals/add-new-product-attribute.md [hunk 8]
```diff
@@ -337,3 +337,54 @@ backend model has executed successfully, so now we’ll set it to Wool and save
...
+1.  You want a set of options which cannot be modified by a user through the admin panel and which can only be changed through a future code push.
+1.  You want a set of options which can be modified, added or deleted through the admin panel.
+
+For use case `1` (an 'immutable' set of options), follow the above instructions ["Add a source model"](#AddSourceModel). You will create a model that contains and dynamically returns the attribute's selectable options to the client.
+
+For use case `2` (a 'mutable' set of options), see ["EAV and extension attributes"]({{ site.baseurl }}/guides/v2.4/extension-dev-guide/attributes.html). Make sure to declare 'Magento\Eav\Model\Entity\Attribute\Source\Table' as the value for the 'source' attribute option. This ensures that Magento will store options in the appropriate database table.
+
+With `\Magento\Eav\Setup\EavSetup.php::addAttribute()` and `\Magento\Eav\Setup\EavSetup.php::addAttributeOptions()` you can add a series of options with the following array:
+
+```
```
Comment (BarnyShergold) [inline src/_videos/fundamentals/add-new-product-attribute.md:358]:
```suggestion
```php
```

File: src/_videos/fundamentals/add-new-product-attribute.md [hunk 9]
```diff
@@ -337,3 +337,54 @@ backend model has executed successfully, so now we’ll set it to Wool and save
...
+'option' => ['values' => ['Option 1', 'Option 2', 'Option 3', etc.]]
+```
+
+Alternatively, you may designate a specific option sorting order as follows:
+
+```
```
Comment (BarnyShergold) [inline src/_videos/fundamentals/add-new-product-attribute.md:364]:
```suggestion
```php
```

File: src/_videos/fundamentals/add-new-product-attribute.md [hunk 10]
```diff
@@ -337,3 +337,54 @@ backend model has executed successfully, so now we’ll set it to Wool and save
...
+
+### Add options to an existing product attribute
+
+*  To add options to an 'immutable' set of options, modify the custom source model with the additional options you wish to provide.
+
+*  Adding options to a 'mutable' set of options leverages the same `EavSetup` object as you use when creating an attribute with options, but requires an additional step because `EavSetup` needs to know to which attribute you want to assign new options.
+
+1. Assign an array of new options to a variable:
+
+```
```
Comment (BarnyShergold) [inline src/_videos/fundamentals/add-new-product-attribute.md:376]:
```suggestion
```php
```

File: src/_videos/fundamentals/add-new-product-attribute.md [hunk 11]
```diff
@@ -337,3 +337,54 @@ backend model has executed successfully, so now we’ll set it to Wool and save
...
+   $options = ['attribute_id' => null, 'values' => 'Option 1', 'Option 2', etc]];
+```
+
+1. Update your array with the attribute ID from the database:
+
+```
```
Comment (BarnyShergold) [inline src/_videos/fundamentals/add-new-product-attribute.md:382]:
```suggestion
```php
```

File: src/_videos/fundamentals/add-new-product-attribute.md [hunk 12]
```diff
@@ -337,3 +337,54 @@ backend model has executed successfully, so now we’ll set it to Wool and save
...
+   $options['attribute_id'] = $eavSetup->getAttributeId($eavSetup->getEntityTypeId('catalog_product'), 'your_attribute_code');
+```
+
+1. Add your options:
+
+```
```
Comment (BarnyShergold) [inline src/_videos/fundamentals/add-new-product-attribute.md:388]:
```suggestion
```php
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Please also check linting - it failed last time for some reason (not clear)

**Comment by dineshvb** [comment]
Hi @BarnyShergold , is this good to be approved?

**Review by BarnyShergold — APPROVED** [review]

**Comment by dineshvb** [comment]
Hi @rogyar , is this good to be approved?

**Review by dobooth — APPROVED** [review]

**Comment by dobooth** [comment]
running tests

**Pull request merged**